pkg/api/norman/customization/aks/handler.go

```diff
@@ -126,6 +140,52 @@ func (h *handler) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
 	}
 }
 
+func (h *handler) checkCredentials(req *http.Request) (int, error) {
+	cred := &Capabilities{}
+	raw, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		return http.StatusBadRequest, fmt.Errorf("cannot read request body: %v", err)
+	}
+
+	if err = json.Unmarshal(raw, &cred); err != nil {
+		return http.StatusBadRequest, fmt.Errorf("cannot parse request body: %v", err)
+	}
+
+	if cred.SubscriptionID == "" {
+		return http.StatusBadRequest, fmt.Errorf("must provide subscriptionId")
+	}
+	if cred.ClientID == "" {
+		return http.StatusBadRequest, fmt.Errorf("must provide clientId")
+	}
+	if cred.ClientSecret == "" {
+		return http.StatusBadRequest, fmt.Errorf("must provide clientSecret")
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	if cred.TenantID == "" {
+		cred.TenantID, err = azureutil.FindTenantID(ctx, azure.PublicCloud, cred.SubscriptionID)
+		if err != nil {
+			return http.StatusBadRequest, err
+		}
+	}
+	if cred.BaseURL == "" {
+		cred.BaseURL = azure.PublicCloud.ResourceManagerEndpoint
```

Comment:
One potential downfall of this whole optional tenant system is azure gov cloud. But we can cross that bridge when we get to it.